packages/ui-link/src/Link/style.js

```diff
@@ -0,0 +1,157 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import generateComponentTheme from './theme'
+/**
+ * Generates the style object from the theme and provided additional information
+ * @param  {Object} theme The actual theme object.
+ * @param  {Object} themeOverride User provided overrides of the default theme mapping.
+ * @param  {Object} props the props of the component, the style is applied to
+ * @param  {Object} state the state of the component, the style is applied to
+ * @return {Object} The final style object, which will be used in the component
+ */
+const generateStyle = (theme, themeOverride, props, state) => {
+  // get the theme variables object for the component
+  const componentTheme = generateComponentTheme(theme, themeOverride)
+  const { isWithinText, renderIcon, iconPlacement, variant, color } = props
+  const inverseStyle = variant === 'inverse' || color === 'link-inverse'
+  const textDecorationStyle = isWithinText
+    ? componentTheme.textDecorationWithinText
+    : componentTheme.textDecorationOutsideText
+  // If Link is a button or link, it should look clickable
+  const buttonOrLinkStyle =
+    state.elementName === 'button' || state.elementName === 'a'
+      ? {
+          cursor: 'pointer',
+          color: inverseStyle
+            ? componentTheme.colorInverse
+            : componentTheme.color,
+          textDecoration: props.isWithinText
+            ? componentTheme.textDecorationWithinText
+            : componentTheme.textDecorationOutsideText,
+          '&:focus': {
+            color: inverseStyle
+              ? componentTheme.colorInverse
+              : componentTheme.color
+          },
+          '&:hover, &:active': {
+            color: inverseStyle
+              ? componentTheme.colorInverse
+              : componentTheme.hoverColor,
+            textDecoration: props.isWithinText
+              ? componentTheme.hoverTextDecorationWithinText
+              : componentTheme.hoverTextDecorationOutsideText
+          }
+        }
+      : {}
+
+  const inverseStyleLinkSelectors = inverseStyle
+    ? {
+        '&:link, &:visited': {
+          color: componentTheme.colorInverse
+        },
+        '&:link:focus, &:visited:focus': {
+          outlineColor: componentTheme.focusInverseIconOutlineColor
+        },
+        '&:link:hover, &:link:focus, &:link:active, &:visited:hover, &:visited:focus, &:visited:active': {
+          color: componentTheme.colorInverse
+        }
+      }
+    : {}
```

Comment:
I'm not sure that this is needed..

Reply:
How else would you apply these style rules?

Reply:
My concern is, that it doesn't exactly translates to the rules in the CSS file, but I suppose it needs VRT and manual testing to see for sure. In the css file, many rules might be there to strengthen the selectors.